src/WmContentManager.php

```diff
@@ -203,33 +199,38 @@ public function getHostContainers($host)
     {
         $containers = $this->getContainers();
 
+        $trigger = $host->getEntityTypeId() . "|" . $host->bundle();
+
+        // Holder.
         $return = [];
 
+        // Go through all containers.
         foreach ($containers as $container) {
-            if (!$container->isHost($host)) {
-                continue;
-            }
+            // Look at the host bundles.
+            $typebundles = $container->host->host_typebundles;
 
-            $return[] = $container->getId();
+            // Go through all of them and if we are there give back that.
+            foreach ($typebundles as $typebundle) {
+                if ($typebundle == $trigger) {
+                    $return[] = $container;
+                }
+            }
         }
 
         return $return;
     }
-    
+
     /**
      * {@inheritdoc}
      */
     public function getLabel($entityType, $bundle)
     {
-        $labels = &drupal_static(__FUNCTION__);
-        if (!isset($labels[$entityType])) {
-            $labels[$entityType] = $this->entityManager->getBundleInfo($entityType);
-        }
-        
-        if (!empty($labels[$entityType][$bundle]['label'])) {
-            return $labels[$entityType][$bundle]['label'];
+        // If there is no selection then all need to be there.
+        $bundles = $this->entityTypeBundleInfo->getBundleInfo($entityType);
+        if (isset($bundles[$bundle]['label'])) {
+            return $bundles[$bundle]['label'];
         }
-        return $bundle;
+        return ucwords(str_replace("_", " ", $bundle));
```

Comment:
@brownrl I would stick to the bundle machine name. This will create confusion for the correct machine name, no?